griptape/tools/web_search/tool.py

```diff
@@ -31,9 +31,11 @@ class WebSearch(BaseTool):
         },
     )
     def search(self, props: dict) -> ListArtifact | ErrorArtifact:
-        query = props["values"]["query"]
+        values = props["values"]
+        query = values["query"]
+        extra_keys = {k: values[k] for k in values.keys() - {"query"}}
```

Comment:
Should the extra keys be put in a separate field from `values` so that we don't need to do this?

Reply:
i think it still makes sense to have everything in `values`

Reply:
could update the decorator to automatically unwrap `values` from `params`. since it looks like every tool has this pattern.
`def search(self, props: dict, values: dict) -> ListArtifact | ErrorArtifact:`

Reply:
I don't want to change the method signature for Tool Activities, that will break all existing Tools. I'm sure we could inspect the method parameters but that could get messy.

I think we either change the behavior so extra values are added in `props["extra_values"]` or we do this.